### DISCUSSION TITLE
Cannot dispatch event to a full page component (using $dispatchTo)

### DISCUSSION THREAD

**Discussion by MUHAMMADSiyab** [livewire/livewire#8568] [7. Bugs (archived)]
### Livewire version

v3.5.0

### Laravel version

v10.47.0

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Dispatch an event to a full page livewire component (using `$dispatchTo` method)

`    <button wire:click="$dispatchTo('demo-page', 'something-happened')">Click Me</button>
`

2. Listen for that event in that full page component

` #[On(['something-happened'])]
    public function reactToEvent() {
        dump("Something Happened!!!");
    }
}`

**Expected Behavior** :
Event should be dispatched and should dump "Something Happened"

**Actual Behavior**:
Nothing is happening

### Repository or Wirebox URL

https://wirebox.app/b/4ko6z

### Additional information

If we use our full page component somewhere in the app like this:

`<livewire:demo-page />
`
The event will be dispatched and everything will work (but we dont need full page component to be used unnecessarily)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by BVFI** [comment]
[This issue](https://github.com/livewire/livewire/discussions/6222) could be related

**Reply by MUHAMMADSiyab** [reply]
Tried the ways explained in this but didn't help

**Reply by BVFI** [reply]
> Tried thw ways explained in this but didn't help

Oh, are you looking for a solution? In that case, could you give more details of your code and what you are trying to do? 
The first alternative that comes to mind is that you could make the ```wire:click=""``` to call a Livewire Component method and then use Livewire's ```dispatch()->to()``` to call that event in that method

**Reply by MUHAMMADSiyab** [reply]
`wire:click` fires the method of component itself, but even `dispatch()->to()` in the component method still results in the same behavior. If we add the full page component like this `<livewire:demo-page/>` , it works

**Reply by BVFI** [reply]
> `wire:click` fires the method of component itself, but even `dispatch()->to()` in the component method still results in the same behavior. If we add the full page component like this `<livewire:demo-page/>` , it works

So, you would have tested it like ```$this->dispatch('something-happened')->to(DemoPage::class);```, correct? I have a ```dispatch()->to()``` in a full page livewire component and it works for me. But since yours isn't that would change behaviors. So you would have this, right:
```php
use \Path\To\DemoPageComponent\DemoPage;
public function event()
{
    $this->dispatch('something-happened')->to(DemoPage::class);
}
```
On a ```<div wire:click="event">``` and it doesn't work? 

Have you put some debug points to see where exactly it stops working?
Can you tell me if it even goes in the method with a Log or Debugbar? 
Not sure about this, but maybe because it's a ```button``` element. Have you tried it with a ```<div>``` one? Or maybe just a [.prevent](https://livewire.laravel.com/docs/wire-click#using-wireclick-on-links)?

**Reply by MUHAMMADSiyab** [reply]
I actually did but till function (event), it works. but at this point 

$this->dispatch()->to 

it stops working

**Comment by hasan-aftab** [comment]
public function redirectUserTo()
    {
        session()->flash('password-updated');
        return $this->redirect(route('login'), navigate: true);
    }

    <script>
        document.addEventListener('livewire:navigated', function () {
            @if(session()->has('password-updated'))
            console.log('password-updated');
            setTimeout(function () {
                Livewire.dispatch('password-updated');
            }, 1000);
            @endif
        });
    </script>
